### DISCUSSION TITLE
为什么查询语句加不加 limit 的执行时间是一样的？

### DISCUSSION THREAD

**Discussion by LCehoennardo** [apache/doris#11807] [A - General / Q&A] [answered]
我尝试了多个查询SQL，发现不管有没有加 limit 分页去限制输出的数据条数，也即不管是输出全量数据还是部分数据，查询时间基本都是一样的，为什么呢，按理说加上 limit 不应该会快一点吗？求解答🙏

**Comment by LCehoennardo** [comment]
doris 版本是 1.0.0

**Comment by LemonLiTree** [comment, marked as answer]
我理解limit只是限制返回的行数，对sql本身对查询可能没有太大的优化帮助，而且数据量不大的话效果也不会明显。可能的情况就是可以避免一些全表扫描的情况，如果sql本身就不需要全表扫描，那也不会有效果。

**Reply by LCehoennardo** [reply]
理解了 多谢
